keyboards/epomaker/th66/keymaps/via/keymap.c

```diff
@@ -0,0 +1,216 @@
+                LCA(EE_CLR), GUI_TOG, WIN_MOD,                            _______,                   _______, _______,          _______, RGB_SPD, RGB_VAD, RGB_SPI
+            ),
+};
+
+void housekeeping_task_user(void) {
+    if (siri_timer_buffer && sync_timer_elapsed32(siri_timer_buffer) > 500) {
+        siri_timer_buffer = 0;
+        unregister_code(KC_LGUI);
+        unregister_code(KC_SPACE);
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case KC_MCTL:
+            if (record->event.pressed) {
+                host_consumer_send(0x29F);
+            } else {
+                host_consumer_send(0);
+            }
+            return false;
+        case KC_SIRI:
+            if (record->event.pressed && siri_timer_buffer == 0) {
+                register_code(KC_LGUI);
+                register_code(KC_SPACE);
+                siri_timer_buffer = sync_timer_read32() | 1;
+            }
+            return false;
+        case WIN_MOD:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_BASE_WIN);
+            }
+            return false;
+        case MAC_MOD:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_BASE_MAC);
+            }
+            return false;
+        case RGB_TOG:
+            if (record->event.pressed) {
+                switch (rgb_matrix_get_flags()) {
+                    case LED_FLAG_ALL: {
+                        rgb_matrix_set_flags(LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER | LED_FLAG_INDICATOR);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    }
+                    break;
+                    case (LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER | LED_FLAG_INDICATOR): {
+                        rgb_matrix_set_flags(LED_FLAG_UNDERGLOW);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    }
+                    break;
+                    case (LED_FLAG_UNDERGLOW): {
+                        rgb_matrix_set_flags(LED_FLAG_NONE);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    }
+                    break;
+                    default: {
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                        rgb_matrix_enable_noeeprom();
+                    }
+                    break;
+                }
+            }
+            return false;
+	}
+
+    return true;
+}
+
+ void rgb_matrix_indicators_user(void) {
+
+    if ((rgb_matrix_get_flags() & LED_FLAG_KEYLIGHT)) {
+        if (host_keyboard_led_state().caps_lock) {
+            rgb_matrix_set_color(31, 255, 0, 0);
+        }
+    } else {
+        if (host_keyboard_led_state().caps_lock) {
+            rgb_matrix_set_color(31, 255, 0, 0);
+        } else {
+            rgb_matrix_set_color(31, 0, 0, 0);
+        }
+    }
+}
```

Comment:
```suggestion
 bool rgb_matrix_indicators_user(void) {
    if ((rgb_matrix_get_flags() & LED_FLAG_KEYLIGHT)) {
        if (host_keyboard_led_state().caps_lock) {
            rgb_matrix_set_color(31, 255, 0, 0);
        }
    } else {
        if (host_keyboard_led_state().caps_lock) {
            rgb_matrix_set_color(31, 255, 0, 0);
        } else {
            rgb_matrix_set_color(31, 0, 0, 0);
        }
    }
    return false;
}
```
Refer to https://github.com/qmk/qmk_firmware/pull/18448